mlflow/_doctor.py

```diff
@@ -102,10 +102,17 @@ def doctor(mask_envs=False):
                 yaml.dump({"_": mlflow_envs}, indent=2).replace("'", "").lstrip("_:").rstrip("\n"),
             )
         )
-    mlflow_dependencies = {
-        r.name: pkg_resources.get_distribution(r.name).version
-        for r in pkg_resources.working_set.by_key["mlflow"].requires()
-    }
```

Comment:
/ai rewrite this use importlib_metadata instead of pkg_resources

Reply:
/ai thanks, but got this error:


```
Traceback (most recent call last):
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/importlib_metadata/__init__.py", line 383, in from_name
    return next(cls.discover(name=name))
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "a.py", line 3, in <module>
    mlflow_dependencies = {dep: version(dep) for dep in requires("mlflow")}
  File "a.py", line 3, in <dictcomp>
    mlflow_dependencies = {dep: version(dep) for dep in requires("mlflow")}
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/importlib_metadata/__init__.py", line 878, in version
    return distribution(distribution_name).version
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/importlib_metadata/__init__.py", line 851, in distribution
    return Distribution.from_name(distribution_name)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/importlib_metadata/__init__.py", line 385, in from_name
    raise PackageNotFoundError(name)
importlib_metadata.PackageNotFoundError: No package metadata was found for click<9,>=7.0
```